src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -619,7 +682,60 @@ public override SingleNamespaceOrTypeDeclaration VisitEnumDeclaration(EnumDeclar
                 nameLocation: new SourceLocation(node.Identifier),
                 memberNames: memberNames,
                 children: ImmutableArray<SingleTypeDeclaration>.Empty,
-                diagnostics: diagnostics.ToReadOnlyAndFree());
+                diagnostics: diagnostics.ToReadOnlyAndFree(),
+                _nonGlobalAliasedQuickAttributes | quickAttributes);
+        }
+
+        private static QuickAttributes GetQuickAttributes(SyntaxList<AttributeListSyntax> attributeLists)
+        {
+            var result = QuickAttributes.None;
+            foreach (var attributeList in attributeLists)
+            {
+                foreach (var attribute in attributeList.Attributes)
+                    result |= GetQuickAttributes(attribute.Name.GetUnqualifiedName().Identifier.ValueText, inAttribute: true);
+            }
+
+            return result;
+        }
+
+        private static QuickAttributes GetQuickAttributes(string name, bool inAttribute)
+        {
+            var result = QuickAttributes.None;
+            if (Matches(AttributeDescription.TypeIdentifierAttribute))
+                result |= QuickAttributes.TypeIdentifier;
+
+            if (Matches(AttributeDescription.TypeForwardedToAttribute))
```

Comment:
Is it worth any effort to treat assembly-only attributes differently than attributes which can be applied to types, etc.? At a glance it looks like only TypeIdentifierAttribute really requires searching the type declarations--the rest can only be applied to assemblies.